plugins/modules/ufw.py

```diff
@@ -581,16 +625,32 @@ def ufw_version():
                             changed = True
                     elif pre_rules != rules_dry:
                         changed = True
+                else:
+                    diff = None
+
+    if not module.check_mode:
+        post_state = execute([[ufw_bin], ['status'], ['verbose']])
+        post_rules = get_current_rules()
+        if (not changed):
+            diff = dict(
+                before="{}\n\n---\n\n{}".format(pre_state, pre_rules),
+                after="{}\n\n---\n\n{}".format(post_state, post_rules),
+            )
+            changed = (pre_state != post_state) or (pre_rules != post_rules)
+
+    if (diff is not None) and (diff["before"] == diff["after"]):
+        diff = None
+
+    if (changed) and (diff is None):
+        raise Exception("ufw module reported a change, but no differences detected")
+    elif (changed is False) and (diff is not None):
```

Comment:
It's unusual to use these many brackets in Python:
```suggestion
    if diff is not None and diff["before"] == diff["after"]:
        diff = None

    if changed and diff is None:
        raise Exception("ufw module reported a change, but no differences detected")
    elif changed is False and diff is not None:
```